net/instaweb/rewriter/strip_subresource_hints_filter_test.cc

```diff
@@ -213,8 +233,10 @@ TEST_F(StripSubresourceHintsFilterTest, SingleResourceDisallow) {
   ValidateStripSubresourceHint(source, rewritten);
 }
 
-TEST_F(StripSubresourceHintsFilterTestPartialPreserve,
-        MultiResourcePreserveAll) {
+// Even if you turn on preserve images, we still strip all rel=subresource hints
+// because we don't know which ones are images.
+TEST_F(StripSubresourceHintsFilterTestPreserveImage,
+       MultiSubresourcePreserveImages) {
   const char *source =
     "<head>"
     "<link rel=\"subresource\" src=\"/dontdropme.gif\"/>"
```

Comment:
Shouldn't you turn all these src attributes into href? Otherwise your "preserve" tests are kinda worthless.